src/common/snippets/src/lowered/pass/assign_registers.cpp

```diff
@@ -16,34 +16,64 @@ namespace snippets {
 namespace lowered {
 namespace pass {
 
+void AssignRegisters::set_reg_types(LinearIR& linear_ir) {
+    for (const auto& expr : linear_ir) {
+        const auto op = expr->get_node();
+        if (ov::is_type<op::LoopEnd>(op) ||
+            ov::is_type<ov::op::v0::Result>(op)
+#ifdef SNIPPETS_DEBUG_CAPS
+        || ov::is_type<op::PerfCountBeginBase>(op)
+        || ov::is_type<op::PerfCountEndBase>(op)
+#endif
+        )
+        continue;
+
+        OPENVINO_ASSERT(expr->get_output_count() == op->get_output_size(), "Incorrect count of output port descriptors!");
```

Comment:
This is pain 😭 
I can't suggest a better solution until we implement typed expressions 😞

Reply:
Totally agree 😢 
I hope that it's really temporary solution until we have typed expressions